File: src/main/java/org/mitallast/queue/raft/resource/ResourceFSM.java
package org.mitallast.queue.raft.resource;

import javaslang.control.Option;
import org.mitallast.queue.common.codec.Message;
import org.mitallast.queue.raft.protocol.RaftSnapshotMetadata;

public interface ResourceFSM {
    Option<Message> prepareSnapshot(RaftSnapshotMetadata snapshotMeta);
}

File: src/main/java/org/mitallast/queue/raft/resource/ResourceFSM.kt
package org.mitallast.queue.raft.resource

import io.vavr.control.Option
import org.mitallast.queue.common.codec.Message
import org.mitallast.queue.raft.protocol.RaftSnapshotMetadata

interface ResourceFSM {
    fun prepareSnapshot(snapshotMeta: RaftSnapshotMetadata): Option<Message>
}
